### PR TITLE
How to install different Unity version

### PR DESCRIPTION
**Pull request by nevercode-kevin**

### CODE REVIEW

File: content/knowledge-base/install-unity-version.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+---
+description: How to install a different Unity
+title: Installing a different Unity version
+weight: 13
+---
+
+# Installing different Unity versions
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:7]: This should be ##

File: content/knowledge-base/install-unity-version.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+Each build machine image has a specific version of Unity installed. You can find out the specific Unity version by consulting the build machine specification in the documentation. For example, the [macOS build machine specification (Xcode 13.3+ / Unity)](https://docs.codemagic.io/specs/versions-macos-xcode-13-3/#unity-2020331f1) has Unity version version 2020.3.31f1.
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:8]: I wouldn't mention any specific Unity versions as these are likely to change sooner or later and we'll probably forget to update this doc here. Maybe there's some other way you can link to the build machine docs.

File: content/knowledge-base/install-unity-version.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+If you need to use a different Unity version then it is possible to use the Unity Hub CLI to download and install a different Unity Editor version and target support files for that version.
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:10]:
```suggestion
If you need to use a different Unity version, then it is possible to use the Unity Hub CLI to download and install a different Unity Editor version and target support files for that version. 
```

Comment (helinanever) [reply content/knowledge-base/install-unity-version.md:10]: is Unity Hub CLI preinstalled?

File: content/knowledge-base/install-unity-version.md (hunk 2 above)
Comment (helinanever) [reply content/knowledge-base/install-unity-version.md:8]: Perhaps link to the most recent image per platform like this: Each build machine image for macOS, Windows, or Linux has a specific version of Unity installed.

File: content/knowledge-base/install-unity-version.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+If you need to use a different Unity version then it is possible to use the Unity Hub CLI to download and install a different Unity Editor version and target support files for that version. 
+
+License activation and return takes place with the Unity version already installed on the machine, but building of the Xcode project or Android binary will use the version of Unity you install. 
+
+
+## Getting the Unity version number and changeset id
+
+In order to install a different version you will need to know the full Unity version number and its **changeset** id. These details can be found in the Unity download archive [here](https://unity3d.com/get-unity/download/archive). To get the changeset id, first select the version you intend to use and then click on the ‘Release Notes’ button. Scroll to the bottom of the release notes and you will see the changeset id which you should make a note of.
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:17]:
```suggestion
In order to install a different version, you will need to know the full Unity version number and its **changeset** id. These details can be found in the Unity download archive [here](https://unity3d.com/get-unity/download/archive). To get the changeset id, first select the version you intend to use and then click on the ‘Release Notes’ button. Scroll to the bottom of the release notes and you will see the changeset id which you should make a note of. 
```

File: content/knowledge-base/install-unity-version.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+
+## Getting the Unity version number and changeset id
+
+In order to install a different version you will need to know the full Unity version number and its **changeset** id. These details can be found in the Unity download archive [here](https://unity3d.com/get-unity/download/archive). To get the changeset id, first select the version you intend to use and then click on the ‘Release Notes’ button. Scroll to the bottom of the release notes and you will see the changeset id which you should make a note of. 
+
+
+## Environment variables
+
+You will need to a three environment variables in your codeamagic.yaml called  `UNITY_VERSION`, `UNITY_VERSION_CHANGSET`, and `UNITY_VERSION_BIN`.
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:22]:
```suggestion
You will need three environment variables in your codemagic.yaml called `UNITY_VERSION`, `UNITY_VERSION_CHANGESET`, and `UNITY_VERSION_BIN`. 
```

File: content/knowledge-base/install-unity-version.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+## Getting the Unity version number and changeset id
+
+In order to install a different version you will need to know the full Unity version number and its **changeset** id. These details can be found in the Unity download archive [here](https://unity3d.com/get-unity/download/archive). To get the changeset id, first select the version you intend to use and then click on the ‘Release Notes’ button. Scroll to the bottom of the release notes and you will see the changeset id which you should make a note of. 
+
+
+## Environment variables
+
+You will need to a three environment variables in your codeamagic.yaml called  `UNITY_VERSION`, `UNITY_VERSION_CHANGSET`, and `UNITY_VERSION_BIN`. 
+
+You should set the value of `UNITY_VERSION` to the full version number as display on the Unity Hub download archive page. For example, if the version is 2019.4.38f1, set the variable as follows:
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:24]:
```suggestion
You should set the value of `UNITY_VERSION` to the full version number as displayed on the Unity Hub download archive page. For example, if the version is `2019.4.38f1`, set the variable as follows:
```

File: content/knowledge-base/install-unity-version.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+`UNITY_VERSION: 2019.4.38f1`
+
+The value of `UNITY_VERSION_CHANGESET` should be set using the **changeset** id that you obtained from the release notes page on the Unity Hub download archive. For example, if the changeset id is fdbb7325fa47 then set the variable as follows:
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:28]:
```suggestion
The value of `UNITY_VERSION_CHANGESET` should be set using the **changeset** id that you obtained from the release notes page on the Unity Hub download archive. For example, if the changeset id is `fdbb7325fa47`, then set the variable as follows:
```

File: content/knowledge-base/install-unity-version.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+`UNITY_VERSION_CHANGESET: fdbb7325fa47`
+
+The `UNITY_VERSION_BIN` should be set as follows so the Unity binary path is declared for the version you want to build with:
+
+`UNITY_VERSION_BIN: /Applications/Unity/Hub/Editor/${UNITY_VERSION}/Unity.app/Contents/MacOS/Unity`
+
+
+## Activating Unity
+
+Even though you are installing a different version of Unity to build your apps with, you should use the default Unity version that is already installed on the machine. Unity Hub CLI commands not work correctly if a license is not already active on the machine.
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:39]: This section needs to be clarified a bit. ... you should activate your license using the default Unity version already installed on the machine? .. activate the default Unity version?

**Review by helinanever — CHANGES_REQUESTED** [review]

File: content/knowledge-base/install-unity-version.md (hunk 8 above)
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:39]:
```suggestion
Unity Hub CLI commands do not work correctly if a license is not already active on the machine.
```

File: content/knowledge-base/install-unity-version.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+
+## Install Unity script
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:42]:
```suggestion
## Unity installation script
```

File: content/knowledge-base/install-unity-version.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+```yaml
+workflows:
+  unity-android-workflow:
+      name: Unity Android Workflow
+      instance_type: mac_pro
+      max_build_duration: 120
+      environment:
+        groups:
+          # Add the group environment variables in Codemagic UI (either in Application/Team variables) - https://docs.codemagic.io/variables/environment-variable-groups/
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:74]:
```suggestion
          # Add the group environment variables in Codemagic UI (in Application or Team variables) - https://docs.codemagic.io/variables/environment-variable-groups/
```

I've always had a problem with this sentence as it's missing the "or" in the either/or construct.

**Review by helinanever — CHANGES_REQUESTED** [review]

File: content/knowledge-base/install-unity-version.md (hunk 3 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md:10]: Yes, it's part of the Unity Hub.

File: content/knowledge-base/install-unity-version.md (hunk 1 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md]: Changed.

File: content/knowledge-base/install-unity-version.md (hunk 2 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md]: Will remove the link to specific versions.

File: content/knowledge-base/install-unity-version.md (hunk 8 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md]: You need to activate the license using the version already installed on the machine. Is this better:


## Activating Unity

Even though you are installing a different version of Unity to build your apps with, you should activate your license using the default Unity version already installed on the machine. Unity Hub CLI commands not work correctly if a license is not already active on the machine.

File: content/knowledge-base/install-unity-version.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+If you need to use a different Unity version, then it is possible to use the Unity Hub CLI to download and install a different Unity Editor version and target support files for that version. 
+
+License activation and return takes place with the Unity version already installed on the machine, but building of the Xcode project or Android binary will use the version of Unity you install. 
+
+
+## Getting the Unity version number and changeset id
+
+In order to install a different version, you will need to know the full Unity version number and its **changeset** id. These details can be found in the Unity download archive [here](https://unity3d.com/get-unity/download/archive). To get the changeset id, first select the version you intend to use and then click on the ‘Release Notes’ button. Scroll to the bottom of the release notes and you will see the changeset id which you should make a note of. 
+
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:18]: unnecessary empty line

File: content/knowledge-base/install-unity-version.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+`UNITY_VERSION: 2019.4.38f1`
+
+The value of `UNITY_VERSION_CHANGESET` should be set using the **changeset** id that you obtained from the release notes page on the Unity Hub download archive. For example, if the changeset id is `fdbb7325fa47`, then set the variable as follows:
+
+`UNITY_VERSION_CHANGESET: fdbb7325fa47`
+
+The `UNITY_VERSION_BIN` should be set as follows so the Unity binary path is declared for the version you want to build with:
+
+`UNITY_VERSION_BIN: /Applications/Unity/Hub/Editor/${UNITY_VERSION}/Unity.app/Contents/MacOS/Unity`
+
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:35]: here too

File: content/knowledge-base/install-unity-version.md [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+## Activating Unity
+
+Unity Hub CLI commands do not work correctly if a license is not already active on the machine.
+
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:40]: and here

File: content/knowledge-base/install-unity-version.md (hunk 8 above)
Comment (helinanever) [reply content/knowledge-base/install-unity-version.md]: That would work

File: content/knowledge-base/install-unity-version.md [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+After activating the Unity license as usual, add the following script to install the desired version and modules you wish to use. The example below uses Unity Hub CLI commands to install the specified Unity version as well as the Android and iOS Build Support modules.
+
+```yaml
+    - name: Install Unity version
+      script: | 
+        /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub -- --headless install --version $UNITY_VERSION --changeset $UNITY_VERSION_CHANGESET 
+        /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub -- --headless install-modules --version $UNITY_VERSION -m ios android 
+```
+
```
Comment (helinanever) [inline content/knowledge-base/install-unity-version.md:52]: and here

File: content/knowledge-base/install-unity-version.md (hunk 8 above)
Comment (helinanever) [reply content/knowledge-base/install-unity-version.md]: Still, one missing word:
Unity Hub CLI commands **do** not work correctly if a license is not already active on the machine.

**Review by helinanever — APPROVED** [review]
Approved with nits :)

File: content/knowledge-base/install-unity-version.md (hunk 8 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md]: corrected

File: content/knowledge-base/install-unity-version.md (hunk 11 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md:17]: sorted

File: content/knowledge-base/install-unity-version.md (hunk 12 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md:33]: removed

File: content/knowledge-base/install-unity-version.md (hunk 13 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md:37]: removed

File: content/knowledge-base/install-unity-version.md (hunk 14 above)
Comment (nevercode-kevin) [reply content/knowledge-base/install-unity-version.md:48]: removed

**Pull request merged**